lib/index.d.ts

```diff
@@ -286,13 +286,6 @@ declare namespace $RefParser {
      */
     order?: number
 
-    /**
-     * All of the built-in parsers allow empty files by default. The JSON and YAML parsers will parse empty files as `undefined`. The text parser will parse empty files as an empty string. The binary parser will parse empty files as an empty byte array.
-     *
-     * You can set `allowEmpty: false` on any parser, which will cause an error to be thrown if a file empty.
-     */
-    allowEmpty?: boolean
```

Comment:
Why do we need to get rid of this? Unlikely to be accepted upstream without good reason, since it's breaking.

Reply:
I got rid of it, because it simply doesn't make sense anymore, since we don't throw anymore when parsing doesn't succeed anyway, so that option wouldn't have any effect.

Reply:
Should we instead treat this as an error is tracked/added to new errors response property if file is empty? So a user can still act on empty files if they want?

Reply:
Honestly, why wouldn't we add a boolean like so:

`continueOnError?: boolean`

To resolver options? The only problem with that is how do add this for local refs `#/definitions/foo`. Right now, at least in the exporter, thats the only time an error is thrown, we catch errors for file/http resolvers.

Reply:
🤷‍♂ we could do that too I suppose - that seems like a separate thing from this `allowEmpty` thing though. Unless we need `continueOnError: false`, I think we don't spend time on that right now.

Reply:
Are we planning on maintaining this fork or merging?

If we are maintaining it, then it's fine. If not, then I think we shouldn't change the existing behavior or the lib and add a new property to allow parsing errors.

Reply:
We can't get away without changing behavior - just trying to minimize it.

Reply:
I am good with whatever, but last suggestion to keep the current behavior intact, and extending it in a way that is useful for everyone, and lastly not having to maintain our own fork.

Suggest we add an onError/on(similar to `process.on`) method. I don't know if it should be on `$RefParser.ParserOptions` or `$RefParser.Options` or both. I am leaning towards revert the change above, and add this method to `$RefParser.Options`, that way it's handled for all errors and isn't specific to parsers. Pretty sure if we put it on `ParserOptions`, we would need to figure out how to handle local JSON Ref error(`#/definitions/does-not-exist`), when they don't exist because we can't set a parser for them.

The below examples are assuming we extend `$RefParser.Options`.

To keep the normal behavior, we wouldn't set an on error function, we would just use a default one, or if we decided to pass in a function we would just throw inside of it.

```ts
const result = await parser.dereference(schema, {
  onError: (/** some useful args for document + position (like our diagnostics), error message, context, whatever */) => {
    // what occurs now, the default behavior, fails fast...
    throw new Error(/** message */)
  }
```

To not fail fast(what we want)

```ts
const errors = [];
const result = await parser.dereference(schema, {
  onError: (/** some useful args for document + position (like our diagnostics), error message, context, whatever */) => {
    // what we want
     errors.push(...whatever)
    // since no error re-thrown here, resolve process continues on...
  }

```

Reply:
I'm fine with adding `continueOnError` flag.
The reason I didn't do it _yet_ is that I don't know how much "error handling code" will be needed in the end, so I just find it easier to let handle all kind of errors.
We can always throw later if needed.